src/utils/ClassHelpers.js

```diff
@@ -0,0 +1,40 @@
+import fastdom from 'fastdom';
+import addOneClass from 'dom-helpers/class/addClass';
+import removeOneClass from 'dom-helpers/class/removeClass';
+
+export function addClass(node, classes) {
+  mutateClass(node, classes, addOneClass);
+}
+export function removeClass(node, classes) {
+  mutateClass(node, classes, removeOneClass);
+}
+
+function mutateClass(node, classes, fn) {
+  if (!node) return;
+  if (classes && typeof classes === 'string') {
+    const run = () => {
+      // A reflow is necessary to get the browser to respect the transition. However, it doesn't
+      // need to be done on every single class change, only when the 'active' class is added.
+      if (classes.indexOf('active') !== -1) {
+        forceReflow(node);
+      }
```

Comment:
I'm actually against doing this here since it makes the class helpers seem very specific to that `active` prefix being there. I think you _can_ change the prefix or set custom CSS classes to apply for?

Doing it in `onEntering` and `onExiting` just like the previous `reflowAndAddClass` function usage seems like the safer option to me. Plus since it's the only place where the `active` classes are going to be added I don't think we're going to miss anything. Plus it's safer when the logic flow is pretty much similar, so we don't end up causing breaking changes to anyone else.

Reply:
You're right about the `active` bit, I've fixed in the latest commit.

In my testing it's only needed on `onEntering`.

Using `fastdom.measure()` does the reflow in a different tick and doesn't show the animation.